### PR TITLE
Implement function that take schema and produce Go code integration

### LINKED ISSUE #1140: Implement function that take schema and produce Go code integration

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #1140

### CODE REVIEW

File: scan/src/bindings/Borsh.re [hunk 1]
```diff
@@ -1,3 +1,6 @@
+[@bs.module "change-case"] external pascalCase: string => string = "pascalCase";
```
Comment (sorawit) [inline scan/src/bindings/Borsh.re]: Have these in its own file.

File: scan/src/bindings/Borsh.re [hunk 2]
```diff
@@ -182,8 +185,81 @@ library ResultDecoder {
...
+	if err != nil {
+		return Result{}, err
+	}|j}
+  };
+};
+
+let resultGo = ({name, varType}) => {
+  let capitalizedName = name |> pascalCase;
+  switch (varType) {
+  | String => {j|$capitalizedName:      $name|j}
```
Comment (sorawit) [inline scan/src/bindings/Borsh.re]: What is this arbitrary spacing?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/bindings/Borsh.re (hunk 2 above)
Comment (taobun) [inline scan/src/bindings/Borsh.re]: Why spaces between words are difference?

File: scan/src/bindings/Borsh.re [hunk 3]
```diff
@@ -182,8 +185,81 @@ library ResultDecoder {
...
+  switch (varType) {
+  | String => {j|	$name, err := decoder.DecodeString()
+	if err != nil {
+		return Result{}, err
+	}|j}
+  | U64 => {j|	$name, err := decoder.DecodeU64()
+	if err != nil {
+		return Result{}, err
+	}|j}
+  | U32 => {j|	$name, err := decoder.DecodeU32()
```
Comment (taobun) [inline scan/src/bindings/Borsh.re]: ditto

File: scan/src/bindings/Borsh.re [hunk 4]
```diff
@@ -182,8 +185,81 @@ library ResultDecoder {
...
+  let template = (structs, functions, results) => {j|package $packageName
+
+type Result struct {
+\t$structs
+}
+
+func DecodeResult(data []byte) (Result, error) {
+\tdecoder := NewBorshDecoder(data)
+
+$functions
```
Comment (taobun) [inline scan/src/bindings/Borsh.re]: If every code correctly, I think we need \t here

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/bindings/Borsh.re [hunk 5]
```diff
@@ -182,8 +182,81 @@ library ResultDecoder {
...
+  | U8 => {j|$name, err := decoder.DecodeU8()
+	if err != nil {
+		return Result{}, err
+	}|j}
+  };
+};
+
+let resultGo = ({name, varType}) => {
+  let capitalizedName = name |> ChangeCase.pascalCase;
+  switch (varType) {
```
Comment (taobun) [inline scan/src/bindings/Borsh.re]: Every type render the same pattern, why we need  to switch

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**